torch_geometric/transforms/pad.py

```diff
@@ -0,0 +1,548 @@
+import numbers
+from collections import defaultdict
+from functools import singledispatchmethod
+from typing import Dict, Generator, List, Optional, Tuple, Union
+
+import torch
+import torch.nn.functional as F
+
+from torch_geometric.data import Data, HeteroData
+from torch_geometric.data.datapipes import functional_transform
+from torch_geometric.transforms import BaseTransform
+
+
+class _AttributeTypeCache:
```

Comment:
Is that such an expensive op? Can we consider moving this to the `Data` class directly?

Reply:
Moved to the `GlobalStorage` class, see [the comment](https://github.com/pyg-team/pytorch_geometric/pull/5940#issuecomment-1326719344) about the performance reasons.